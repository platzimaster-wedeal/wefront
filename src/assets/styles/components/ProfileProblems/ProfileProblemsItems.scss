@import "../../partials/colors.scss";
@import "../../partials/fonts.scss";
@import "../../FontSize.scss";
@import "../../partials/text.scss";
@import "../../partials/sizes.scss";

.ProfileProblems__item {
  display: flex;
  align-items: center;
  margin-bottom: 20px;
 
  &:last-child {
   margin-bottom: 0;
  }
 }

.ProfileProblems__item--icon {
  color: $color-black-90;
  margin-right: 22px;
  @include font-size($font-size-big);

  @include media(mobile-2) {
    @include font-size($font-size-normal );
   }
  
   @include media(mobile-l) {
    @include font-size($font-size-medium);
   }
  
   @include media(tablet) {
    @include font-size($font-size-big);
   }
 }

.ProfileProblems__id {
  color: $color-black-90;
  font-style: normal;
  font-weight: 500;
  text-decoration: underline;
  letter-spacing: 0.02em;
  @include font-size($font-size-medium);
 
  @include media(mobile-2) {
   @include font-size($font-size-small );
  }
 
  @include media(mobile-l) {
   @include font-size($font-size-normal);
  }
 
  @include media(tablet) {
   @include font-size($font-size-medium);
  }
 }
 