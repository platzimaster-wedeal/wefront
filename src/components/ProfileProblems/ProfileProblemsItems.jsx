import React from "react";
import { Link } from "react-router-dom";
import { MdErrorOutline } from "react-icons/md";
import "../../assets/styles/components/ProfileProblems/ProfileProblemsItems.scss";

const ProfileProblemsItems = ({ problemId = "1" }) => {
 return (
  <Link className="ProfileProblems__item" to={`/detail/problem/${problemId}`}>
   <MdErrorOutline className="ProfileProblems__item--icon" />
   <p className="ProfileProblems__id">#{problemId}</p>
  </Link>
 );
};

export default ProfileProblemsItems;
