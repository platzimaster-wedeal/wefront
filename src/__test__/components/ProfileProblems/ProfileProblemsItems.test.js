import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import ProfileProblemsItems from "../../../components/ProfileProblems/ProfileProblemsItems";

describe("<Profile Problems Items/>", () => {
 const profileProblemsItems = mount(<ProfileProblemsItems />);
 test("Render del componente ProfileProblemsItems", () => {
  expect(profileProblemsItems.length).toEqual(1);
 });
 test("Render del tÃ­tulo de un componente", () => {
  expect(profileProblemsItems.find(".ProfileProblems__id").text()).toEqual(
   "#"
  );
 });
});

describe("Profile Problems Items Snapshot", () => {
 test("Comprobar la UI del componente ProfileProblemsItems", () => {
  const profileProblemsItems = create(<ProfileProblemsItems />);
  expect(profileProblemsItems.toJSON()).toMatchSnapshot();
 });
});