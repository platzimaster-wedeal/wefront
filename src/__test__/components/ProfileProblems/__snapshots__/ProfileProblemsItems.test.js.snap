// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Problems Items Snapshot Comprobar la UI del componente ProfileProblemsItems 1`] = `
<div
  className="ProfileProblems__item"
>
  <svg
    className="ProfileProblems__item--icon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
    />
  </svg>
  <p
    className="ProfileProblems__id"
  >
    #
  </p>
</div>
`;
